src/qililab/platform/components/bus.py

```diff
@@ -197,49 +180,78 @@ def get_parameter(self, parameter: Parameter, channel_id: int | None = None):
         """
         if parameter == Parameter.DELAY:
             return self.settings.delay
-        try:
-            return self.system_control.get_parameter(
-                parameter=parameter, channel_id=channel_id, port_id=self.port, bus_alias=self.alias
-            )
-        except ParameterNotFound as error:
-            raise ParameterNotFound(
-                f"No parameter with name {parameter.value} was found in the bus with alias {self.alias}"
-            ) from error
+        for instrument, instrument_channel in zip(self.instruments, self.channels):
+            with contextlib.suppress(ParameterNotFound):
+                if channel_id is not None and channel_id == instrument_channel:
+                    return instrument.get_parameter(parameter, channel_id)
+                return instrument.get_parameter(parameter, instrument_channel)
+        raise Exception(f"No parameter with name {parameter.value} was found in the bus with alias {self.alias}")
 
     def upload_qpysequence(self, qpysequence: QpySequence):
         """Uploads the qpysequence into the instrument."""
-        self.system_control.upload_qpysequence(qpysequence=qpysequence, port=self.port)
+        from qililab.instruments.qblox.qblox_module import QbloxModule  # pylint: disable=import-outside-toplevel
+
+        for instrument, instrument_channel in zip(self.instruments, self.channels):
+            if isinstance(instrument, QbloxModule):
+                instrument.upload_qpysequence(qpysequence=qpysequence, channel_id=int(instrument_channel))  # type: ignore[arg-type]
+                return
+
+        raise AttributeError(f"Bus {self.alias} doesn't have any QbloxModule to upload a qpysequence.")
 
     def upload(self):
         """Uploads any previously compiled program into the instrument."""
-        self.system_control.upload(port=self.port)
+        for instrument, instrument_channel in zip(self.instruments, self.channels):
+            if isinstance(instrument, (QbloxQCM, QbloxQRM)):
+                instrument.upload(channel_id=instrument_channel)
+                return
 
     def run(self) -> None:
         """Runs any previously uploaded program into the instrument."""
-        self.system_control.run(port=self.port)
+        for instrument, instrument_channel in zip(self.instruments, self.channels):
+            if isinstance(instrument, (QbloxQCM, QbloxQRM)):
```

Comment:
Why here in bus we only use Qblox? I don't understand what happens with QM

Reply:
This is a method that is executed only when executing a qibo circuit which is supported in qblox only